lib/galaxy/workflow/errors.py

```diff
@@ -0,0 +1,198 @@
+"""
+Functionality for sending error reports for workflow runs.
+"""
+
+import string
+
+import markupsafe
+
+from galaxy import (
+    model,
+    util,
+)
+from galaxy.security.validate_user_input import validate_email_str
+from galaxy.util import unicodify
+
+error_report_template = """
+GALAXY WORKFLOW RUN ERROR REPORT
+------------------------
+
+This error report was sent from the Galaxy instance hosted on the server
+"${host}"
+-----------------------------------------------------------------------------
+This is in reference to workflow invocation id ${invocation_id} (${invocation_id_encoded}) from history id ${history_id} (${history_id_encoded})
+-----------------------------------------------------------------------------
+You should be able to view the history associated with the workflow invocation (${invocation_id_encoded}) for workflow
+
+${workflow_name}
+
+by logging in as a Galaxy admin user to the Galaxy instance referenced above
+and pointing your browser to the following link.
+
+${history_view_link}
+-----------------------------------------------------------------------------
+The user ${email_str} provided the following information:
+
+${message}
+-----------------------------------------------------------------------------
+(This is an automated message).
+"""
+
+error_report_template_html = """
+<html>
+    <body>
+<h1>Galaxy Workflow Run Error Report</h1>
+<span class="sub"><i>from</i> <span style="font-family: monospace;"><a href="${host}">${host}</a></span>
+
+<h3>Error Localization</h3>
+<table style="margin:1em">
+    <tbody>
+        <tr><td>Workflow</td><td><a href="${workflow_view_link}">${workflow_name} (${workflow_id}) (${workflow_id_encoded})</a></td></tr>
```

Comment:
Don't include the decoded id here. The email may be responded to, and I think is also delivered to users ?

Reply:
Good catch. 

Should I also remove the decoded id from the dataset/job error email, or can that be done separately?

Reply:
I suppose a separate PR is better ? You probably wouldn't expect that change in this PR.